fix : Improve map fragment

Shailesh351 opened the pull request:
Fixes #1743 

Changes: 
- Fixed padding before autocomplete search view
- Changed autocomplete textview to SearchView.SearchAutoComplete to make UI consistent
- Showed marker info on clicking on suggestion
- Added null check for event and location
- Removed sharing option from fDroid version
- Clicking map icon from sesssion detail activity and location activity directly focus on corresponding location and shows marker info.

Screenshots for the change: 

![videotogif_2017 06 29_18 48 32](https://user-images.githubusercontent.com/18264684/27689797-ff4e9b4c-5cfc-11e7-8408-bada856e556b.gif)

Shailesh351 commented:
@arp95 @shreyasnbhat @iamareebjamal @rohanagarwal94 Please review.

Shailesh351 commented:
Which one will be better  ?

![videotogif_2017 06 29_19 08 35](https://user-images.githubusercontent.com/18264684/27690307-9ad4d85a-5cfe-11e7-8b6e-7d7346173ed3.gif)

ghost commented:
Hi @Shailesh351!

It looks like one or more of your builds have failed.
I've added the relevant info below to save you some time.

- [Build 3505.1](https://travis-ci.org/fossasia/open-event-android/jobs/248371496) has `failed`

shreyasnbhat commented:
@Shailesh351 I think that the 2nd one would be better

arp95 commented:
i think the same. 2nd will be better.

Shailesh351 commented:
Do not merge this PR. Some work is remaining. Working on it

ghost commented:
Hi @Shailesh351!

It looks like one or more of your builds have failed.
I've added the relevant info below to save you some time.

- [Build 3556.1](https://travis-ci.org/fossasia/open-event-android/jobs/249175946) has `failed`

Shailesh351 commented:
Guys now clicking map icon from sesssion detail activity and location activity directly focus on corresponding location and shows marker info of that location.

Ready to merge.
@iamareebjamal @arp95 @shreyasnbhat @rohanagarwal94 Please review now.

Shailesh351 commented:
![videotogif_2017 07 02_01 25 20](https://user-images.githubusercontent.com/18264684/27765038-9f44b6ee-5ec5-11e7-8a12-6e5486eebf41.gif)

iamareebjamal commented on android/app/src/fdroid/java/org/fossasia/openevent/fragments/OSMapFragment.java:
```diff
@@ -223,22 +217,6 @@ public void onDestroy() {
-        return super.onOptionsItemSelected(item);
-    }
-
 //    private void launchDirections() {
```
Can you remove the comments too? Thanks

Shailesh351 committed:
fix : Improve map fragment

Fixed padding before autocomplete search view

Changed autocomplete textview to SearchView.SearchAutoComplete

Showed marker info on clicking on suggestion

Showed marker info on clicking map icon from session detail activity and location activity
